singer_sdk/mapper.py

```diff
@@ -326,7 +338,8 @@ def _eval(
         Raises:
             MapExpressionError: If the mapping expression failed to evaluate.
         """
-        names = record.copy()  # Start with names from record properties
+        names = self.module_names  # Start with the module names
+        names.update(record.copy())  # Add top-level record properties
```

Comment:
If the record has a field named `json` it'd override the `json` module. I also don't think we need all the members of `json`, just `dumps` and `loads`.

Reply:
We could add them to the functions mapping:

```python
funcs["json.loads"] = json.loads
funcs["json.dumps"] = json.dumps
```